metis-indexing/src/main/java/eu/europeana/indexing/FullBeanPublisher.java

```diff
@@ -207,16 +203,29 @@ private void publish(RdfWrapper rdf, Date recordDate, List<String> datasetIdsToR
 
     // Publish to Solr
     try {
-      ExternalRequestUtil.retryableExternalRequest(() -> {
-            publishToSolr(rdf, savedFullBean);
-            return null;
-          }, Collections.singletonMap(UnknownHostException.class, ""), PUBLISH_MAX_RETRIES,
-          PERIOD_BETWEEN_RETRIES_IN_MILLIS);
-    } catch (IndexingException e) {
-      throw e;
+      retryableExternalRequestForNetworkExceptions(() -> {
+        try {
+          publishToSolr(rdf, savedFullBean);
+        } catch (IndexingException e) {
+          throw new RuntimeException(e);
+        }
+        return null;
```

Comment:
Perhaps instead of catching this exception, it is better to rework the retryable method to use `SupplierThrowingException<R>` instead of a regular supplier?

Reply:
I did that originally but then I realized that was too much overhead which I could avoid by simply doing this here, which not much different if not at all..
What do you think?

Reply:
I think that perhaps eventually we may want to do this to make our utils code more generic. But we can perhaps leave that for the next time.

Reply:
Ok, let's see how that sinks in first then